### PR TITLE
Use installed libraries

### PR DESCRIPTION
**Pull request by jedwards4b**
This change allows cesm to use installed PIO libraries should they exist on a system.  An installed PIO library implies that the following items are defined in the environment:
PIO_VERSION_MAJOR, PIO_LIBDIR, PIO_INCDIR, PIO_TYPENAME_VALID_VALUES
I have installed pio modules on cheyenne for several combinations of compiler and mpilib.

Test suite: scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: config/cesm/machines/config_machines.xml [hunk 1]
```diff
@@ -289,7 +289,7 @@ This allows using a different mpirun command to launch unit tests
     <NODENAME_REGEX>.*.cheyenne.ucar.edu</NODENAME_REGEX>
     <!-- MPT sometimes timesout at model start time, the next two lines cause
     case_run.py to detect the timeout and retry FORCE_SPARE_NODES times -->
-    <MPIRUN_RETRY_REGEX>MPT: xmpi_net_accept_timeo/accept() timeout</MPIRUN_RETRY_REGEX>
+    <MPIRUN_RETRY_REGEX>MPT: Launcher network accept (MPI_LAUNCH_TIMEOUT) timed out</MPIRUN_RETRY_REGEX>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: This is an unrelated machine change - with the change of mpt on cheyenne the message text has changed.

File: src/build_scripts/buildlib.pio [hunk 2]
```diff
@@ -47,6 +47,19 @@ def buildlib(bldroot, installpath, case):
...
+    # If variable PIO_VERSION_MAJOR is defined in the environment then
+    # we assume that PIO is installed on the system
+    # and expect to find
+    # PIO_LIBDIR, PIO_INCDIR, PIO_TYPENAME_VALID_VALUES
+    # also defined in the environment.  In this case we
+    # will use the installed pio and not build it here.
+    installed_pio_version = os.environ.get("PIO_VERSION_MAJOR")
+    logger.info("pio_version_major = {} pio_version = {}".format(installed_pio_version, pio_version))
+    if installed_pio_version is not None and int(installed_pio_version) == pio_version:
+        logger.info("Using installed PIO library")
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply src/build_scripts/buildlib.pio]: It's included in the pio.bldlog file - which log file are you thinking of?

[the remark this answers is not in the document]
Comment (fischer-ncar) [reply src/build_scripts/buildlib.pio]: The same log file as the "Using installed PIO library" is going to.  Even changing the message
to "Using installed PIO library, version 1" would be fine.

[the remark this answers is not in the document]
Comment (jedwards4b) [reply src/build_scripts/buildlib.pio]: It's there - in the line above pio_version_major = {} pio_version = {} . pio_version is what you are using.

[the remark this answers is not in the document]
Comment (fischer-ncar) [reply src/build_scripts/buildlib.pio]: I was too focused on the "Using installed PIO library line".  It's fine the way you have it.
